internal/service/core/core_instance_resource.go

```diff
@@ -1621,6 +1621,9 @@ func (s *CoreInstanceResourceCrud) SetData() error {
 
 	if s.Res.ExtendedMetadata != nil {
 		s.D.Set("extended_metadata", tfresource.GenericMapToJsonMap(s.Res.ExtendedMetadata))
+	} else {
+		extended_metadata := map[string]interface{}{}
+		s.D.Set("extended_metadata", extended_metadata)
```

Comment:
Why does setting the values to an empty map fix the issue you are facing?

Reply:
Hi thanks for reviewing. 

## Problem Description
At the moment we are having trouble with aggregate data types like maps being returned as null by the PCA API. When stored as `null` in the state file these aggregate data type variables report a change as being required every plan despite the values not being specified in the terraform. 

To give a concrete example, below I have an image of the state file for vm instance created with the OCI Terraform provider on a Private Cloud Appliance. You can see that the value for system tags is set to `null` which is due to the behavior of the PCA API.
![image](https://github.com/oracle/terraform-provider-oci/assets/86663659/fbff4f67-5a3d-446e-870e-fabfdd89ad19)
*Image 1: system_tags set to null*


Whenever this `system_tags` variable is set to `null` in the state file a `terraform plan` will always report `(known after apply)` as the value it will be changed to. Terraform will always detect changes to be made despite no real changes in a given terraform codebase. This is okay when dealing with a single vm instance. However larger scales these reported changes become very problematic and make it very difficult to determine legitimate drift in your configuration. 
![image](https://github.com/oracle/terraform-provider-oci/assets/86663659/8b67aae5-a70d-430a-a0f9-bdbdf3f00612)
*Image 2: terraform plan reports the `system_tags` value as requiring a change despite its value being unspecified.*

We can fix this false `system_tags` change by setting it to an empty map `{}` through a manual edit of the state file. Below I have an image of the state file after being manually edited. 
![image](https://github.com/oracle/terraform-provider-oci/assets/86663659/5d6a388c-9890-4e3d-8c17-e2ddd206d750)
*Image 3: system_tags is manually set to {} in the state file.*

The resulting `terraform plan` no longer reports the `system_tags` attribute as requiring a change as can be seen in Image 4.
![image](https://github.com/oracle/terraform-provider-oci/assets/86663659/b81a37b0-6ea1-4b3a-8c87-d4bb56b91fc6)
*Image 4: There is no required change detected for system_tags.*

We can fix the `extended_metadata` attribute in the same way. By changing it in the state file from `null` to `{}` this will remove the false change from the plan entirely. 
![image](https://github.com/oracle/terraform-provider-oci/assets/86663659/24e61da4-e0d0-45b8-9ddc-ca639310817d)
*Image 5: After setting the values in the state file from null to {} no change is detected.*

I suspect for those without a PCA to access you will be able to confirm this problematic behavior from `null` by manually editing a state file for a vm instance on OCI so that its `system_tags` attribute is set to `null`.